Source/CMakeLists.txt

```diff
@@ -20,9 +20,12 @@ else()
   # CMAKE_CXX_STANDARD cannot be used here because we require C++14 or newer, not any standard.
   check_and_add_flag(CXX17 -std=c++17)
   if(NOT FLAG_CXX_CXX17)
-    set(CMAKE_CXX_STANDARD 14)
-    set(CMAKE_CXX_STANDARD_REQUIRED ON)
-    set(CMAKE_CXX_EXTENSIONS OFF)
+  	check_and_add_flag(CXX1Z -std=c++1z)
```

Comment:
Is it possible to implement this logic only for android?

Reply:
I'm not sure what the cleanest way of doing that in the CMake file would be, but I'll give it a shot